File: 240221/방향에 맞춰 이동/move-in-direction.py
N = int(input())
dx, dy = [1,0,-1,0], [0,-1,0,1]
x, y = 0, 0

for i in range(N) : 
  a, b = input().split()
  b = int(b)
  if a == 'N':
    y = y + dy[3] * b
  elif a == 'E' :
    x = x + dx[0] * b
  elif a == 'S':
    y = y + dy[1] * b
  elif a == 'W' :
    x = x + dx[2] * b

print(x, y)

File: 240410/방향에 맞춰 이동/move-in-direction.js
const fs = require('fs')
const input = fs.readFileSync(0).toString().trim().split(`\n`)
const n = Number(input[0])

let commands = []

for (let i=1; i <= n; i++) {
    commands.push(input[i].split(' '))
}

let x = 0, y =0;

for (let i=0; i < n; i++) {
    if (commands[i][0] === 'N') {
        y += parseInt(commands[i][1])
    }
    else if (commands[i][0] === 'E') {
        x += parseInt(commands[i][1])
    }
    else if (commands[i][0] === 'S') {
        y -= parseInt(commands[i][1])
    }
    else if (commands[i][0] === 'W') {
        x -= parseInt(commands[i][1])
    }
}

console.log(x, y)